src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -61,17 +61,26 @@ public class CompilationWithAnalyzers
         /// An optional cancellation token which can be used to cancel analysis.
         /// Note: This token is only used if the API invoked to get diagnostics doesn't provide a cancellation token.
         /// </summary>
-        public CancellationToken CancellationToken => _cancellationToken;
+        [Obsolete("This CancellationToken is always 'default'", error: false)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public CancellationToken CancellationToken => default;
```

Comment:
Technically `None` and `default` are the same, but `None` is easier for reading.

```suggestion
        [Obsolete("This CancellationToken is always 'None'", error: false)]
        [EditorBrowsable(EditorBrowsableState.Never)]
        public CancellationToken CancellationToken => CancellationToken.None;
```